packages/@aws-cdk/aws-sns/lib/subscription-filter-policy.ts

```diff
@@ -0,0 +1,150 @@
+abstract class Filter {
+  /**
+   * The conditions of the filter.
+   * Conditions are `OR`ed.
+   */
+  public readonly conditions: any[] = [];
+}
+
+/**
+ * Filter for a string attribute.
+ */
+export class StringFilter extends Filter {
+  /**
+   * Match one or more values.
+   * Can be chained with other conditions.
+   */
+  public whitelist(...values: string[]) {
```

Comment:
We've decided we're going to try to not do fluent interfaces anymore. Would something like:

```ts
Filter.stringFilter({
  whitelist: [...],
  blacklist: [...],
  prefixes: [...]
})
```

etc also work?

Reply:
Do you want to preserve the `addXxx()` methods ?

```ts
const filterPolicy = new sns.SubscriptionFilterPolicy();
filterPolicy.addStringFilter('color').whitelist('red', 'orange').matchPrefixes('bl');
filterPolicy.addStringFilter('size').blacklist('small', 'medium');
filterPolicy.addNumericFilter('price').between(100, 200).greaterThan(300);
```
would become:
```ts
const filterPolicy = new sns.SubscriptionFilterPolicy();
filterPolicy.addStringFilter('color', {
  whitelist: ['red', 'orange'],
  matchPrefixes: ['bl'],
});
filterPolicy.addStringFilter('size', {
  blacklist: ['small', 'medium']
});
filterPolicy.addNumericFilter('price', {
  between: [100, 200],
  greatThan: 300
});
```

Another option without the `addXxx()` methods is:
```ts
const filterPolicy = new sns.SubscriptionFilterPolicy({
  color: Filter.stringFilter({
    whitelist: ['red', 'orange'],
    matchPrefixes: ['bl'],    
  }),
  size: Filter.stringFilter({
    blacklist: ['small', 'medium']
  }),
  price: Filter.numericFilter({
    between: [100, 200],
    greatThan: 300
  })
});
```